### PR TITLE
HPA support for the various services

### PR DESCRIPTION
**Pull request by ermeaney**

### CODE REVIEW

File: templates/server-configmap.yaml [hunk 1]
```diff
@@ -77,10 +77,8 @@ data:
                 url:
                     scheme: "{{ $.Values.elasticsearch.scheme }}"
                     host: "{{ $.Values.elasticsearch.host }}:{{ $.Values.elasticsearch.port }}"
-                username: "{{ $.Values.elasticsearch.username }}"
-                password: "{{ $.Values.elasticsearch.password }}"
                 indices:
-                    visibility: "{{ $.Values.elasticsearch.visibilityIndex }}"
+                    visibility: temporal-visibility-dev
```
Comment (chitreshd) [inline templates/server-configmap.yaml:81]: why are we hardcoding?

File: values.yaml [hunk 2]
```diff
@@ -156,6 +162,12 @@ server:
 
   history:
     # replicaCount: 1
+    autoscaling:
+      enabled: false
+      minReplicas: 1
+      maxReplicas: 3
```
Comment (chitreshd) [inline values.yaml:168]: for prod you would need to start with atleast 3 replicas

File: values.yaml [hunk 3]
```diff
@@ -174,6 +186,12 @@ server:
 
   matching:
     # replicaCount: 1
+    autoscaling:
+      enabled: false
+      minReplicas: 1
```
Comment (chitreshd) [inline values.yaml:191]: same as history

File: values.yaml [hunk 4]
```diff
@@ -287,9 +311,6 @@ elasticsearch:
   host: elasticsearch-master-headless
   scheme: http
   port: 9200
-  username: ""
-  password: ""
-  visibilityIndex: "temporal-visibility-dev"
 
 prometheus:
   enabled: true
```
Comment (chitreshd) [inline values.yaml:316]: i thought we wanted to disable prometheus

**Comment by underrun** [comment]
Thanks for the contribution! Please remember to sign the CLA.

For this specific PR, we've recently landed on an alternate approach to helm charts: https://github.com/temporalio/proposals/issues/42 - as a result, and because HPA resources can be added independently of temporal itself, we will not be merging this at this time.

**Pull request closed without merging**